Islandora/islandora
# Deletion abstraction

**GitHub Issue**: ([link](https://github.com/Islandora/documentation/issues/2298))


# What does this Pull Request do?
 Abstracts duplicated code to a function in Utils

# What's new?
The logic for media and file deletion has been moved from i`slandora_object_delete_form_submit` and `ConfirmDeleteMediaAndFile::submitForm` has been moved to `IslandoraUtils::deleteMediaAndFiles
`
# How should this be tested?
Delete an Islandora node, with `Delete all associated medias and nodes` checked
Check to see that the media and files associated with that node have been deleted.
Select `Delete Media and File(s) `from the Action menu and check to see that all expected deletions have occurred.

# Documentation Status

No change in use



# Interested parties
 @Islandora/committers @kayakr
